Source/WebCore/style/values/inline/StyleLineHeight.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <WebCore/StyleLengthWrapper.h>
+
+namespace WebCore {
+namespace Style {
+
+// <'line-height'> = normal | <number [0,∞]> | <length-percentage [0,∞]>
+// NOTE: <number [0,∞]> gets converted to <length-percentage [0,∞]>.
+// https://drafts.csswg.org/css-inline/#propdef-line-height
+struct LineHeight : LengthWrapperBase<LengthPercentage<CSS::Nonnegative>, CSS::Keyword::Normal> {
+    using Base::Base;
+
+    bool isNormal() const { return holdsAlternative<CSS::Keyword::Normal>(); }
+
+    unsigned valueForTextAutosizingHash() const
+    {
+        return switchOn(
+            [](const CSS::Keyword::Normal&) -> unsigned { return 0; },
+            [](const Fixed& fixed) -> unsigned { return static_cast<unsigned>(fixed.resolveZoom(ZoomNeeded { })); },
+            [](const Percentage& percentage) -> unsigned { return static_cast<unsigned>(percentage.value); },
```

Comment:
Consider preserving all the bits since both `float` and `unsigned` are 32 bits?
```suggestion
            [](const Fixed& fixed) -> unsigned { return std::bit_cast<unsigned>(fixed.resolveZoom(ZoomNeeded { })); },
            [](const Percentage& percentage) -> unsigned { return std::bit_cast<unsigned>(percentage.value); },
```
Not sure this is `float`, or if `bit_cast` will even compile.

Do we want to do anything to cut down on collisions between similar values of the 4 different types?

Finally, why would we want to resolve the fixed value to hash it?

Reply:
We shouldn’t resolve them. I’ll fix that.

Reply:
Ideally I should hash-combine with the variant index. Will fix that too.